Sources/HiPMobile/BusinessLayer/ContentApiFetchers/DataToRemoveFetcher.cs

```diff
@@ -168,7 +168,7 @@ private void RemoveRouteTags(Route route, List<RouteTag> deletedTags, List<Image
                 {
                     deletedTags.Add(tag);
                 }
-                if (tag.Image != null && !allMedias.Contains(tag.Image.IdForRestApi))
+                if (tag.Image != null && !allMedias.Contains(tag.Image.IdForRestApi) && tag.Image.IdForRestApi != -2)
```

Comment:
What's that magic `-2` value? Perhaps put it into a named constant?

Reply:
This is default image having id -2, which should not be deleted as it's not part of the data store. Will put this into constant